bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/InputRenderer.java

```diff
@@ -58,38 +74,271 @@ public boolean canRender(Widget w) {
+                }
+            }
+        }
+        snippet = snippet.replace("%data_state%", dataState);
+        snippet = snippet.replace("%item_state%", itemState);
+
+        String unitSnippet = "";
+        String unit = "";
+        if (item instanceof NumberItem) {
+            NumberItem numberItem = (NumberItem) item;
+            if (numberItem.getDimension() != null) {
+                unit = getUnit(w, numberItem);
+                if ("number".equals(inputHint)) {
+                    unitSnippet = "<span %valuestyle% class=\"mdl-form__input-unit\">" + unit + "</span>";
+                }
+            }
+        }
+        snippet = snippet.replace("%item_unit%", unit);
+        snippet = snippet.replace("%unit_snippet%", unitSnippet);
+
         // Process the color tags
         snippet = processColor(w, snippet);
 
         sb.append(snippet);
+
         return ECollections.emptyEList();
     }
+
+    private String parseNumber(String value) {
+        String newValue = value.trim();
+        Matcher numberMatcher = NUMBER_PATTERN.matcher(newValue);
+        if (numberMatcher.find()) {
+            String numberValue = numberMatcher.group(0);
+            String unitValue = newValue.substring(numberValue.length()).trim();
+            newValue = numberValue.replace("/^\\+/", "");
+            if (COMMA_SEPARATOR_PATTERN.matcher(newValue).find()) {
+                newValue = newValue.replace("/\\./g", "").replace(",", ".");
+            }
+            if (unitValue.length() > 0) {
+                newValue = newValue + " " + unitValue;
+            }
+            return newValue;
+        } else {
+            return value;
+        }
+    }
+
+    private String getValue(Widget w, Item item) {
+        String value = cleanValue(getValue(w), w, item);
+        if (value.isBlank()) {
+            State state = itemUIRegistry.getState(w);
+            if (state != null && !(state instanceof UnDefType)) {
+                value = state.toString();
+            } else {
+                value = "-";
+            }
+        }
+        if (item instanceof NumberItem) {
+            NumberItem numberItem = (NumberItem) item;
+            if (numberItem.getDimension() != null) {
+                String[] stateArray = value.split(" ");
+                if (stateArray.length <= 1) {
+                    value = (stateArray.length > 0 ? stateArray[0] : value) + " " + getUnit(w, numberItem);
+                }
+            }
+        }
+        return value;
+    }
+
+    private String cleanValue(String value, Widget w, Item item) {
+        String prefix = getPrefix(w, item);
+        String postfix = getPostfix(w, item);
+        String newValue = value.startsWith(prefix) ? value.substring(prefix.length()) : value;
+        newValue = value.endsWith(postfix) ? newValue.substring(0, newValue.lastIndexOf(postfix)) : newValue;
+        return newValue.trim();
+    }
+
+    private String getUnit(Widget w, NumberItem item) {
```

Comment:
This is the new method that is key to making it work, and was not there before May 22nd.